### PR TITLE
Add call classification and supported kind to data extensions editor

### PR DESCRIPTION
**Pull request by koesie10**
This will add a "Test" or "Generated" tag on the method if it's only used in tests or generated code and not in the actual application source code. It also adds retrieving the method "type" (e.g. source or sink) to the queries, but does not show these yet.

Screenshot 2023-07-13 at 16 46 04

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/view/data-extensions-editor/MethodRow.tsx [hunk 1]
```diff
@@ -197,6 +221,12 @@ function ModelableMethodRow(props: Props) {
           </UsagesButton>
         )}
         <ViewLink onClick={jumpToUsage}>View</ViewLink>
+        {!inSource && (
+          <ClassificationsContainer>
+            {inTest && <VSCodeTag>Test</VSCodeTag>}
```
Comment (robertbrignull) [inline extensions/ql-vscode/src/view/data-extensions-editor/MethodRow.tsx:226]: Random thought. Is it possible to put a tooltip / title text on these that explains "This method is only used from test code"?

**Review by robertbrignull — APPROVED** [review]
No comments on how the code is written. That all looks great.

My only comment on the styling is whether the classification labels are a bit too large and prominent, and if it makes the view more cluttered. It's the default styling so it's definitely not your fault, but they maybe stand out slightly too much and distract from other elements, and it's unhelpful that they look almost exactly the same as the clickable buttons.

However, I'd be happy to proceed as this is now, with the understanding that we'll be doing a design review once we get to the end of this current implementation work.

**Comment by koesie10** [comment]
> My only comment on the styling is whether the classification labels are a bit too large and prominent, and if it makes the view more cluttered. It's the default styling so it's definitely not your fault, but they maybe stand out slightly too much and distract from other elements, and it's unhelpful that they look almost exactly the same as the clickable buttons.

I agree, the tags are a little bit big. I've made their text sizes smaller so they stand out less, so this should hopefully partly fix that. I've also added the tooltip you suggested. Unfortunately, the webview UI toolkit doesn't have a tooltip component, so it's just a `title` on a `<div>` which shows when you hover over the tags for some time.

**Review by robertbrignull — APPROVED** [review]
LGTM

Thanks for those styling tweaks, and putting the classification code in a new component looks good.

**Pull request merged**